src/SmartComponents/DriftPage/DriftTable/ComparisonHeader/SystemHeaderCellContent.js

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useSelector } from 'react-redux';
+import RemoveSystemCell from './RemoveSystemCell';
+import DriftTooltip from '../../../DriftTooltip/DriftTooltip';
+import HistoricalProfilesPopover from '../../../HistoricalProfilesPopover/HistoricalProfilesPopover';
+import { BlueprintIcon, ClockIcon, ExclamationTriangleIcon, ServerIcon } from '@patternfly/react-icons';
+import ReferenceSelector from '../ReferenceSelector/ReferenceSelector';
+import helpers from '../../../helpers';
+
+const SystemHeaderCellContent = ({
+    fetchCompare,
+    item,
+    permissions,
+    referenceId,
+    removeSystemFunc,
+    selectHistoricProfiles,
+    systemIds,
+    updateReferenceId
+}) => {
+    const selectedBaselineIds = useSelector(({ baselinesTableState }) => baselinesTableState.comparisonTable.selectedBaselineIds);
+    const getItemType = (itemType) => {
+        if (itemType === 'system') {
+            return 'System';
+        } else if (itemType === 'baseline') {
+            return 'Baseline';
+        } else if (itemType === 'historical-system-profile') {
+            return 'Historical System';
+        }
+    };
+
+    const getItemIcon = (itemType) => {
+        if (itemType === 'system') {
+            return <ServerIcon/>;
+        } else if (itemType === 'baseline') {
+            return <BlueprintIcon/>;
+        } else if (itemType === 'historical-system-profile') {
+            return <ClockIcon />;
+        }
+    };
+
+    const itemType = getItemType(item.type);
+    const itemIcon = getItemIcon(item.type);
```

Comment:
You can use `useMemo` to define `itemType` and `itemType` and avoid having to define  the functions to call above.